docs/integration-testing.md

```diff
@@ -19,53 +18,66 @@ typically run faster. Sometimes that's not (easily) possible, especially when an
 
 ## Running them locally
 
-The JUnit runner won't pick up integration tests unless the `RUN_INTEGRATION_TEST` environment variable is set to `true`
-. Also, don't forget to define any credentials that are needed.
+As mentioned above the JUnit runner won't pick up integration tests unless a tag is provided. For instance if needed to run integration tests related to `Azure CosmosDB` then it can be achieved by passing the `IncludeTags` parameter to the `gradlew` command:
+
+```bash
+./gradlew test -DIncludeTags="azure-cosmos-db-integration-test"
+```
+
+_Cosmos DB integration tests are run by default against a locally running [Cosmos DB Emulator](https://docs.microsoft.com/azure/cosmos-db/local-emulator). You can also use an instance of Cosmos DB running in Azure, in which case you should set the `COSMOS_KEY` environment variable._
 
-Cosmos DB integration tests are run by default against a locally running [Cosmos DB Emulator](https://docs.microsoft.com/azure/cosmos-db/local-emulator). You can also use an instance of Cosmos DB running in Azure, in which case you should set the `COSMOS_KEY` environment variable.
+if needed to run all kind of tests(e.g. unit & integration) then it can be achieved by passing the `RunAllTest=true` parameter to the `gradlew` command:
+
+```bash
+./gradlew test -DRunAllTest="true"
+```
+
+If needed to run all integration tests from a module (and all of its sub-modules) then it can be achieved by specifying module project to the `gradlew` command. For example to run all integration tests from Azure cosmos db module and its sub-modules:
+
+```bash
+./gradlew -p extensions/azure/cosmos test -DIncludeTags="azure-cosmos-db-integration-test"
+```
+
+_Command as `./gradlew :extensions:azure:cosmos test -DIncludeTags="azure-cosmos-db-integration-test"` does not executed tests from all sub-modules. So we need to use `-p` to specify the module project path._
 
 ## Running them in the CI pipeline
 
-All integration tests should go into the [integration test workflow](../.github/workflows/integrationtests.yaml),
-every "technology" should have its own job, every test should go into a step.
+All integration tests should go into the [integration test workflow](../.github/workflows/integrationtests.yaml), every "technology" should have its own job, and technology specific tests can be targeted using Junit tags with `-DIncludeTags` property as described above in document.
+
+For example let's assume we've implemented a Postgres-based Asset Index, then the integration tests for that should go into a "Postgres" `job`, and every module that adds a test (here: `extensions:postgres:assetindex`) should apply a composite annotation (here: `@PostgresIntegrationTest` adding a tag `postgres-integration-test`) on its integration tests this tagging will be used by the CI pipeline step to target and execute the integration tests related to Postgres.
```

Comment:
not a Postgres job, a step

Reply:
Missing full stop before "this tagging"?

Reply:
- This will be a job. Step is mentioned later in the paragraph ` This tagging will be used by the CI pipeline step to target...`
- Applied the full stop.